lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1567,11 +1567,11 @@ protected function _validateAndCompleteOneToOneMapping(array $mapping)
             foreach ($mapping['joinColumns'] as &$joinColumn) {
                 if ($mapping['type'] === self::ONE_TO_ONE && ! $this->isInheritanceTypeSingleTable()) {
                     if (count($mapping['joinColumns']) == 1) {
-                        if ( ! isset($mapping['id']) || ! $mapping['id']) {
-                            $joinColumn['unique'] = true;
+                        if ( (! isset($mapping['id']) || ! $mapping['id'])) {
+                            $joinColumn['unique'] = isset($joinColumn['unique']) ? $joinColumn['unique'] : true ;
                         }
                     } else {
-                        $uniqueConstraintColumns[] = $joinColumn['name'];
+                        $uniqueContraintColumns[] = $joinColumn['name'];
```

Comment:
Probably because I initially did this on the 2.3 branch, then noticed that the project maintainers prefer everything to be done on master,  and  the variable was incorrectly named on the 2.3 branch.  I can fix that as well.